Create/Modify Trick

guillaumedbk opened the pull request.

guillaumedbk committed:
Controller + Form + Template for trick creation

guillaumedbk committed:
createTrick method with image uploading

guillaumedbk committed:
Placeholder + require for trick creation form

guillaumedbk committed:
Beginning of the modifyTrick method: controller, form & template

guillaumedbk committed:
ModifyTrick form + controller method

cacahouwete commented on src/Controller/TrickController.php:
```diff
@@ -35,4 +43,101 @@ public function __invoke(TricksRepository $tricksRepository, int $id, ChatReposi
+    public function modifyTrick(Request $request, int $id, TricksRepository $tricksRepository, VideoRepository $videoRepository, ImageRepository $imageRepository,  EntityManagerInterface $entityManager): Response
+    {
+        //VALUES
+        $trick = $tricksRepository->find($id);
```
```suggestion
        $trick = $tricksRepository->find($id);
        if ($trick === null) {
            throw new NotFoundHttpException("Not found", null);
        }
```

cacahouwete commented on src/DTO/TrickDTO.php:
```diff
@@ -18,4 +18,7 @@ class TrickDTO
     public $imageUrl;
+
+    #[Assert\Url]
+    public $videoUrl;
```
Maybe you should use array of dto for image and video.
```suggestion
    public $images;
    public $videos;
```

guillaumedbk committed:
deleteTrick method + add links into trick card

guillaumedbk committed:
Retrieve images into the dto

guillaumedbk committed:
Auto stash before rebase of "master"

+ delete imageUrl column

guillaumedbk committed:
add mainImage Trick attribute in one to one relationship with Image

guillaumedbk committed:
ImageFileManager with upload file method + adapt picture display

cacahouwete commented on src/Controller/HomeController.php:
```diff
@@ -11,14 +12,16 @@
+    public function __invoke(TricksRepository $tricksRepository, ImageRepository $imageRepository): Response
     {
         $tricks = $tricksRepository->findAll();
+        $images = $imageRepository->findAll();
```
You should use trick.images in your twig to get images

cacahouwete commented on templates/home/index.html.twig:
```diff
@@ -24,16 +24,28 @@
+                            <a href="{{ path('app_trick', {id: figure.id}) }}">
+                                {% if images %}
+                                    {% for image in images %}
+                                        {% if image.id is same as figure.mainImage.id %}
```
```suggestion
                                        {% if figure.mainImage not null %}
```

cacahouwete commented on templates/home/index.html.twig:
```diff
@@ -24,16 +24,28 @@
+                                {% if images %}
+                                    {% for image in images %}
+                                        {% if image.id is same as figure.mainImage.id %}
+                                            <img src="/uploads/{{ image.url }}" class="card-img-top" alt={{ figure.title }}>
```
```suggestion
                                            <img src="/uploads/{{ figure.mainImage.url }}" class="card-img-top" alt={{ figure.title }}>
```

cacahouwete commented on src/Manager/ImageFileManager.php:
```diff
@@ -0,0 +1,27 @@
+use App\Entity\Trick;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+class ImageFileManager extends AbstractController
```
You have to remove extends AbstractController and create construct with imagesDirectory attribute.

In your services.yaml you have to add binding key for this attr like :

```yaml
services:
    _defaults:
        bind:
            $imagesDirectory: '%.....%'
```

cacahouwete commented on src/Manager/ImageFileManager.php:
```diff
@@ -0,0 +1,27 @@
+        //UPLOAD AND ADD ALL IMAGES
+        foreach ($images as $image){
+            //NEW FILE NAME
+            $file = md5(uniqid()) . '.' . $image->guessExtension();
```
can you rename file to fileName

cacahouwete commented on src/Manager/ImageFileManager.php:
```diff
@@ -0,0 +1,27 @@
+
+class ImageFileManager extends AbstractController
+{
+    public function uploadFile(array $images, Trick $trick)
```
you have to add array type. (and change to iterable if possible)

```php
/**
 * @param iterable<UploadFile> $images
 */
```

cacahouwete commented on src/Form/CreateTrickType.php:
```diff
@@ -0,0 +1,44 @@
+                'attr' => array('placeholder' => 'ex: https://www.youtube.com/embed/'),
+                'required' => false
+            ])
+            ->add('images', FileType::class, [
```
You should use https://symfony.com/doc/current/reference/forms/types/collection.html to handl all images and videos

cacahouwete commented on src/Entity/Trick.php:
```diff
@@ -200,4 +187,16 @@ public function removeImage(Image $image): self
+
+        return $this;
+    }
+
```
public function setMainImageWithFirstImage(): void
    {
        $this->mainImage = $this->images->first();
    }

guillaumedbk committed:
Twig correction for image retrieval and display

guillaumedbk committed:
Twig corrections for retrieving and displaying items

guillaumedbk committed:
Construct with imagesDirectory attribute + bind in services.yaml

guillaumedbk committed:
Rename

guillaumedbk committed:
Typing

guillaumedbk committed:
setMainImage + delete Cascade

retrieve the first image and set it as main image for the Trick

guillaumedbk committed:
CollectionType for videos

CollectionType with VideoType that retrieve url in VideoDTO, possibility to dynamically add fields to add several videos to a trick

guillaumedbk committed:
ImageType, retrieve data in ImageDTO

guillaumedbk committed:
Adapt modify trick

guillaumedbk committed:
deletePicture method

guillaumedbk committed:
Possibility to add comment on trick if connected

guillaumedbk committed:
upload images for fixtures and adapt fixtures

guillaumedbk committed:
Icon to edit and delete on trick page

cacahouwete commented on src/Manager/ImageFileManager.php:
```diff
@@ -0,0 +1,36 @@
+{
+    protected $imageDirectory;
+
+    public function __construct($imageDirectory)
```
```suggestion
    public function __construct(string $imageDirectory)
```

cacahouwete commented on src/Manager/ImageFileManager.php:
```diff
@@ -0,0 +1,36 @@
+
+class ImageFileManager
+{
+    protected $imageDirectory;
```
```suggestion
    private string $imageDirectory;
```

guillaumedbk committed:
fixtures corrections + deleteImage method into modify trick form

guillaumedbk committed:
NotFoundHttpException + New line character

guillaumedbk committed:
check if the user is logged in before being able to create/modify/delete

+ adapt header

guillaumedbk committed:
Flash message

guillaumedbk committed:
Pagination for comments

guillaumedbk committed:
Load More button

guillaumedbk committed:
Hotfix

guillaumedbk committed:
exception handling and uniqueness constraints for figures

guillaumedbk committed:
add lastModified column/attribute for trick

guillaumedbk committed:
URLs of the figure pages contain the name of the figure as a slug

guillaumedbk committed:
strict comparison
